### DISCUSSION TITLE
Members of microbial community have different values for flux of exchanged metabolite.

### DISCUSSION THREAD

**Discussion by sankarmaheshr** [micom-dev/micom#155] [Q&A] [answered]
I'm building a community of 2 microbes. They are identical SBML models with few changes that were introduced to have heterologous reactions.
Product A is secreted by microbe A into the environment. Product A is then taken up by microbe B for further conversions.
As per MICOM, when doing this to get the flux values, I observed that independent of the flux values of product A secreted (2E-6), the flux of product A being taken up by microbe B (100) is much higher.
What could have caused this?
![MICOM error](https://github.com/micom-dev/micom/assets/124859020/d7d85f41-29ad-42a4-a5f7-530d1c51a228)

**Comment by cdiener** [comment]
Hmm, not too much information to go on. What does the flux for `EX_pcoum_m` look like? Did you make sure that `pcoum` is not in your medium?

**Comment by sankarmaheshr** [comment]
Hi. The flux for EX_pcoum in organism 1 is 0.00002 and organism 2 is 100. I checked the medium and it doesnt have any pcoum.

**Comment by cdiener** [comment]
Okay would need to check a few more things, but basically it looks like pcoum is coming from somewhere else. You could try looking at `com.metabolites.pcoum_m.reactions` and `com.metabolites.pcoum_e__taxa2.reactions` to see if those look correct. Also, what does `com.reactions.EX_pcoum_e__taxa2.reaction` look like (to check it is actually import and not export).

**Reply by sankarmaheshr** [reply]
Dear Diener, An update on what I've done so far.
Since I'm interested in building a community where a metabolite is exchanged between 2 organisms, I checked the earlier discussions and followed [https://github.com/micom-dev/micom/discussions/148].
I had included 4 reactions in organism 1,

`reaction = cobra.Reaction("EX_pcoum")`
`Community_1_model_1.add_reactions([reaction])`
`reaction.reaction = " <-> pcoum_e"`

`reaction = cobra.Reaction("pcoum_transport")`
`Community_1_model_1.add_reactions([reaction])`
`reaction.reaction = "pcoum_c <-> pcoum_e"`

`reaction = cobra.Reaction("DM_pcoum_c")`
`Community_1_model_1.add_reactions([reaction])`
`reaction.reaction = "pcoum_c -> "`

`reaction = cobra.Reaction("DM_pcoum_e")`
`Community_1_model_1.add_reactions([reaction])`
`reaction.reaction = "pcoum_e -> "`

And 2 reactions in organism 2

`reaction = cobra.Reaction("EX_pcoum")`
`Community_1_model_2.add_reactions([reaction])`
`reaction.reaction = " <-> pcoum_e"`

`reaction = cobra.Reaction("pcoum_transport")`
`Community_1_model_2.add_reactions([reaction])`
`reaction.reaction = "pcoum_e <-> pcoum_c"`

What I could observe now is that, although the values of pcoum secreted by organism 1 is always higher than the values of pcoum taken by organism 2. I also observed that some pcoum is in the exchange compartment (_m).
But, these do not equate.
So then I checked if there are other reactions that take flux away from organism 2. 
![MICOM flux values not matching](https://github.com/micom-dev/micom/assets/124859020/c4868ef2-1611-4ea0-a220-38c8ef2de537)

Since abundances of both the organisms are equal 0.5, the fluxes should equate.
How can I rationalize the inequality and flux in the exchange compartment?

**Reply by cdiener** [reply, marked as answer]
Okay this looks much better. The fluxes do not have to be the same but they have to be balanced. Imports are weighted by relative abundance so the corresponding balance would be `0.5*EX_pcoum__taxa1 + 0.5*EX_pcoum__taxa2 - EX_pcoum_m = 0`. In your case the difference between the taxa is `0.5*(2.5e-5 - 1.8e-5) = 3.5e-6` which gets rounded to `0.000004` in the display. So that looks correct. Here it just means that some of the `pcoum` gets secreted in the medium and not taken up by organism 2 which can happen especially if is not needed for growth for organism 2 (saturated import for example).

**Reply by sankarmaheshr** [reply]
Thanks for clarifying @cdiener.
